core/src/banking_stage.rs

```diff
@@ -249,17 +252,15 @@ impl BankingStage {
         buffered_packets: &mut Vec<PacketsAndOffsets>,
         enable_forwarding: bool,
     ) -> Result<()> {
-        let (decision, next_leader) = {
+        let decision = {
             let poh = poh_recorder.lock().unwrap();
-            let next_leader = poh.next_slot_leader();
-            (
-                Self::consume_or_forward_packets(
-                    next_leader,
-                    poh.bank().is_some(),
-                    poh.would_be_leader(DEFAULT_TICKS_PER_SLOT * 2),
-                    my_pubkey,
+            Self::consume_or_forward_packets(
+                poh.next_slot_leader(),
+                poh.bank().is_some(),
+                poh.would_be_leader(
+                    (FORWARD_TRANSACTIONS_TO_LEADER_AT_SLOT_OFFSET - 1) * DEFAULT_TICKS_PER_SLOT,
```

Comment:
why is this -1?

Reply:
If the current node will be leader in next N-1 slots...